### PR TITLE
Capture UnauthorizedServiceException by AuditableEnforcer

### PR DESCRIPTION
**Pull request by tsschmidt**
Adds UnauthorizedServiceException to the list of Exceptions captured by RegisteredServiceAccessStrategyAuditableEnforcer  specifically for the cases where a service is disabled in the service registry.  This will allow all calling methods to handle enforcer calls the same by checking the returned result for a captured exception and eliminates the need to code a try/catch block.

### CODE REVIEW

**Comment by mmoayyed** [comment]
> ... for the cases where a service is disabled in the service registry.

Could you add a few test cases to demonstrate this?

**Comment by tsschmidt** [comment]
Will do

File: core/cas-server-core-services-api/src/test/java/org/apereo/cas/services/util/RegisteredServiceAccessStrategyAuditableEnforcerTests.java [hunk 1]
```diff
@@ -0,0 +1,310 @@
...
+    }
+
+    private ServiceTicket createServiceTicket() {
+        val mock = mock(ServiceTicket.class);
+        when(mock.getService()).thenReturn(createService());
+        return mock;
+    }
+
+    private AuthenticationResult createAuthenticationResult() {
+        val mock = mock(AuthenticationResult.class);
```
Comment (mmoayyed) [inline core/cas-server-core-services-api/src/test/java/org/apereo/cas/services/util/RegisteredServiceAccessStrategyAuditableEnforcerTests.java]: Should remove, and use similar functionality in `CoreAuthenticationTestUtils`

File: core/cas-server-core-services-api/src/test/java/org/apereo/cas/services/util/RegisteredServiceAccessStrategyAuditableEnforcerTests.java [hunk 2]
```diff
@@ -0,0 +1,310 @@
...
+    private TicketGrantingTicket createTicketGrantingTicket() {
+        val mock = Mockito.mock(TicketGrantingTicket.class);
+        val authentication = createAuthentication();
+        when(mock.getAuthentication()).thenReturn(authentication);
+        when(mock.isRoot()).thenReturn(true);
+        when(mock.getRoot()).thenReturn(mock);
+        return mock;
+    }
+
+    private ServiceTicket createServiceTicket() {
```
Comment (mmoayyed) [inline core/cas-server-core-services-api/src/test/java/org/apereo/cas/services/util/RegisteredServiceAccessStrategyAuditableEnforcerTests.java]: Should remove, and use similar functionality in `MockServiceTicket`

File: core/cas-server-core-services-api/src/test/java/org/apereo/cas/services/util/RegisteredServiceAccessStrategyAuditableEnforcerTests.java [hunk 3]
```diff
@@ -0,0 +1,310 @@
...
+    private Principal createPrincipal() {
+        return new Principal() {
+            @Override
+            public String getId() {
+                return "principal";
+            }
+        };
+    }
+
+    private TicketGrantingTicket createTicketGrantingTicket() {
```
Comment (mmoayyed) [inline core/cas-server-core-services-api/src/test/java/org/apereo/cas/services/util/RegisteredServiceAccessStrategyAuditableEnforcerTests.java]: Should remove, and use similar functionality in `MockTicketGrantingTicket`

File: core/cas-server-core-services-api/src/test/java/org/apereo/cas/services/util/RegisteredServiceAccessStrategyAuditableEnforcerTests.java [hunk 4]
```diff
@@ -0,0 +1,310 @@
...
+        val principal = createPrincipal();
+        val attributes = new HashMap<String, Object>();
+        attributes.put("attribute", "value");
+        val mock = mock(Authentication.class);
+        when(mock.getAttributes()).thenReturn(attributes);
+        when(mock.getPrincipal()).thenReturn(principal);
+        return mock;
+    }
+
+    private Principal createPrincipal() {
```
Comment (mmoayyed) [inline core/cas-server-core-services-api/src/test/java/org/apereo/cas/services/util/RegisteredServiceAccessStrategyAuditableEnforcerTests.java]: Should remove, and use similar functionality in `CoreAuthenticationTestUtils`

File: core/cas-server-core-services-api/src/test/java/org/apereo/cas/services/util/RegisteredServiceAccessStrategyAuditableEnforcerTests.java [hunk 5]
```diff
@@ -0,0 +1,310 @@
...
+    private Service createService() {
+        return new Service() {
+            @Override
+            public String getId() {
+                return "serviceid";
+            }
+        };
+    }
+
+    private Authentication createAuthentication() {
```
Comment (mmoayyed) [inline core/cas-server-core-services-api/src/test/java/org/apereo/cas/services/util/RegisteredServiceAccessStrategyAuditableEnforcerTests.java]: Should remove, and use similar functionality in `CoreAuthenticationTestUtils`

File: core/cas-server-core-services-api/src/test/java/org/apereo/cas/services/util/RegisteredServiceAccessStrategyAuditableEnforcerTests.java [hunk 6]
```diff
@@ -0,0 +1,310 @@
...
+
+    private RegexRegisteredService createRegisteredService(final boolean enabled) {
+        val service = new RegexRegisteredService();
+        val accessStrategy = new DefaultRegisteredServiceAccessStrategy();
+        accessStrategy.setEnabled(enabled);
+        service.setAccessStrategy(accessStrategy);
+        return service;
+    }
+
+    private Service createService() {
```
Comment (mmoayyed) [inline core/cas-server-core-services-api/src/test/java/org/apereo/cas/services/util/RegisteredServiceAccessStrategyAuditableEnforcerTests.java]: Should remove, and use similar functionality in `CoreAuthenticationTestUtils`

File: core/cas-server-core-services-api/src/test/java/org/apereo/cas/services/util/RegisteredServiceAccessStrategyAuditableEnforcerTests.java [hunk 7]
```diff
@@ -0,0 +1,310 @@
...
+
+    @Test
+    public void verifyExceptionNotThrown() {
+        val context = AuditableContext.builder().build();
+        val result= new RegisteredServiceAccessStrategyAuditableEnforcer().execute(context);
+        assertTrue(result.isExecutionFailure());
+        assertTrue(result.getException().isPresent());
+    }
+
+    private RegexRegisteredService createRegisteredService(final boolean enabled) {
```
Comment (mmoayyed) [inline core/cas-server-core-services-api/src/test/java/org/apereo/cas/services/util/RegisteredServiceAccessStrategyAuditableEnforcerTests.java]: Should remove, and use similar functionality in `CoreAuthenticationTestUtils`

**Pull request merged**